spec/app/ics-028-cross-chain-validation/system_model_and_properties.md

```diff
@@ -128,6 +134,10 @@ CCV provides the following system invariants.
   >
   > **Note:** The *Slash Invariant* also ensures that if a delegator starts unbonding an amount `x` of tokens from `val` before height `hi`, then `x` will not be slashed, since `x` is not part of `Token(Power(c,hi,val))`.
 
+- ***Distribution Invariant***: If a consumer chain sends to the provider chain an amount `T` of tokens as reward for providing security, then
+  - `T` (equivalent) tokens are eventually minted on the provider chain and then distributed among the validators that are part of the validator set;
+  - the total supply of tokens is preserved, i.e., the `T` (original) tokens are escrowed on the consumer chain.
```

Comment:
I think this mixes a liveness property into the invariant:
- liveness: If a consumer chain sends to the provider chain an amount `T` of tokens as reward for providing security, then`T` (equivalent) tokens are eventually minted on the provider chain and then distributed among the validators that are part of the validator set;
- invariant: It is always the case that the total sum of the minted fee tokens on the provider chain is less than or equal to the total sum of the escrowed tokens on the consumer chain.

Reply:
Indeed. Nice catch. I replaced "system invariants" with "system properties", since I believe it's more accurate. Many of the existing "invariants" had hidden temporal properties in them.